code/web/src/modules/survey/Survey.js

```diff
@@ -0,0 +1,120 @@
+// Imports
+import React, { useEffect, useState } from 'react';
+import { APP_URL } from '../../setup/config/env';
+import { connect } from 'react-redux';
+import { withRouter } from 'react-router-dom'
+
+// App Imports
+import { surveyQuestions } from './surveyQuestions';
+import userRoutes from '../../setup/routes/user';
+import { postUserStyle } from './api/actions';
+import { login } from '../user/api/actions';
+
+const Survey = (props) => {
+  const [question, setQuestion] = useState(surveyQuestions[0]);
+  const [userStyle, setUserStyle] = useState(null);
+  const [styleAnswers, setStyleAnswers] = useState({
+    Grunge: [],
+    Sporty: [],
+    Classic: [],
+    Casual: [],
+    Bohemian: [],
+  });
+
+  const recordAnswer = (cat) => {
+    setStyleAnswers(styleAnswers, styleAnswers[cat].push(question.id));
+    let i = surveyQuestions.indexOf(question);
+
+    if (i === surveyQuestions.length - 1) {
+      calculateStyle();
+      return;
+    }
+
+    setQuestion(surveyQuestions[i + 1]);
+  };
+
+  const calculateStyle = () => {
+    let tally = Object.entries(styleAnswers);
```

Comment:
I really like how this is tallying up the entries selected and then sorting them to find the style. It even handles some edge cases. Nice solution!